src/urllib3/response.py

```diff
@@ -701,14 +702,46 @@ def read(
         flush_decoder = False
         fp_closed = getattr(self._fp, "closed", False)
 
+        # Reading more than 2 GiB (`int` max value in C) via SSL on some
+        # CPython 3.8, 3.9, and 3.10 versions leads to an overflow error
+        # that has to be prevented if `amt` or `self.length_remaining`
+        # indicate that it may happen.
+        c_int_max = (2**31) - 1
+        if (
+            (amt and amt > c_int_max)
+            or (self.length_remaining and self.length_remaining > c_int_max)
+        ) and (
+            (3, 8, 0, "a", 4) <= sys.version_info < (3, 9, 7)
+            or (3, 10, 0, "b", 1) <= sys.version_info < (3, 10, 0, "rc", 1)
+        ):
+
+            def read(amt: Optional[int] = None) -> bytes:
+                if self._fp is None:
+                    return b""
+                buffer = io.BytesIO()
+                while amt is None or amt != 0:
+                    if amt is not None:
+                        chunk_amt = min(amt, c_int_max)
+                        amt -= chunk_amt
+                    else:
+                        chunk_amt = c_int_max
+                    data = self._fp.read(chunk_amt)
+                    if not data:
+                        break
+                    buffer.write(data)
+                return buffer.getvalue()
+
+        else:
+            read = self._fp.read
+
         with self._error_catcher():
             if amt is None:
                 # cStringIO doesn't like amt=None
```

Comment:
Can we please update this comment? `cStringIO` was removed in Python 3. (Can be done in a future PR)

Reply:
Let's remove now since we're touching this area and the comment is out of date.

Reply:
I updated "cStringIO" to "StringIO" in 8fff25d, I was not sure if it is good to remove the comment